src/scripts/main.js

```diff
@@ -1,3 +1,35 @@
 'use strict';
 
-// write your code here
+const thead = document.querySelector('thead');
+const tbody = document.querySelector('tbody');
+const tfoot = document.querySelector('tfoot');
+const columnIndex = 1;
+
+// Функція для клонування клітинки і вставки перед останнім стовпцем
+function cloneAndInsertBeforeLast(row) {
+  const cellToClone = row.cells[columnIndex];
+
+  if (cellToClone) {
```

Comment:
This check ensures that the cell to be cloned exists. However, if `columnIndex` is out of bounds for any row, this condition will not prevent an error. Consider adding a check to ensure `columnIndex` is within the valid range for the row.